src/farcasterd/runtime.rs

```diff
@@ -928,90 +927,98 @@ impl Runtime {
-
-                    // Connect
-                    let peer_connected_is_ok =
-                        match (self.connections.contains(&peer), peer_secret_key) {
-                            (false, None) => return self.get_secret(endpoints, source, request),
-                            (false, Some(sk)) => {
-                                debug!(
-                                    "{} to remote peer {}",
-                                    "Connecting".bright_blue_bold(),
-                                    peer.bright_blue_italic()
-                                );
-                                match self.connect_peer(source.clone(), &peer, sk) {
-                                    Err(err) => {
-                                        report_to.push((
-                                            Some(source.clone()),
-                                            Request::Failure(Failure {
-                                                code: 1,
-                                                info: err.to_string(),
-                                            }),
-                                        ));
-                                        false
-                                    }
-                                    Ok(()) => true,
-                                }
-                            }
-                            (true, _) => {
-                                let msg = format!(
-                                    "Already connected to remote peer {}",
-                                    peer.bright_blue_italic()
-                                );
-                                warn!("{}", &msg);
-                                true
-                            }
-                        };
-
-                    if peer_connected_is_ok {
-                        let offer_registered = format!(
-                            "{}: {:#}",
-                            "Public offer registered".bright_green_bold(),
-                            &public_offer.id().bright_yellow_bold()
+                    )?;
+                    return Ok(());
+                }
+                let PublicOffer {
+                    version: _,
+                    offer: _,
+                    node_id,      // bitcoin::Pubkey
+                    peer_address, // InetSocketAddr
+                } = public_offer;
+
+                let daemon_service = internet2::RemoteNodeAddr {
+                    node_id,                                           // checked above
+                    remote_addr: RemoteSocketAddr::Ftcp(peer_address), /* expected RemoteSocketAddr */
+                };
+                let peer = daemon_service
+                    .to_node_addr(LIGHTNING_P2P_DEFAULT_PORT)
+                    .ok_or(internet2::presentation::Error::InvalidEndpoint)?;
+
+                // Connect
+                info!("going twice?????????????");
```

Comment:
Still needed?

Reply:
No, must have snuck back in during rebase. Removed in 0e194d8d37b3cb973ff60176fa58285e02fd226a.